### ISSUE TITLE
opera init: `create.yml` does not exist

### ISSUE THREAD

**Issue by duelle** [xlab-si/xopera-opera#81]
Version: `opera 0.5.8`

When running the `opera init http.zip` command with the provided zip file ([http.zip](https://github.com/xlab-si/xopera-opera/files/4973145/http.zip)), the following error is reported:

```
% opera init http.zip 
.opera/csars/csar/_definitions/radonnodestesting__DeploymentTestAgent.tosca:23:24: [Path] Path /tmp/http/nodetypes/radon.nodes.testing/DeploymentTestAgent/files/create/create.yml does not exist.
```

The file is present in the CSAR/ZIP file but obviously not in the given path as the extraction seems to happen inside the `.opera` folder and not directly to the folder the  `http.zip` file is located in (`/tmp/http/`).

The `create.yml` is available in the `/tmp/http/.opera/csars/csar/nodetypes/radon.nodes.testing/DeploymentTestAgent/files/create` folder after the `init` command though.

**Comment by anzoman** [comment]
HI @duelle thanks for spotting this issue. I tried to run this myself and I get the same result. 

This seems to be firstly because of the working directory you were running the opera commands from. Like if the CSAR would be extracted to `/tmp/http/csar` and then if you run it from `tmp/http` you would get the same error because you are of course not running from the root of the csar. So when you move to `tmp/http/csar` everything works fine. But there is a catch with compressed csars which are (as you found out) extracted to `.opera/csars`. Currently only one CSAR can be used within the same working directory and it gets extracted to `.opera/csars/csar` (so it it also renamed to csar). But when you try to initialized it everything would work if you had your playbooks deeper in the structure than the TOSCA service template. However, the problem occurs when you want to link your playbooks from the root of the CSAR because opera assumes that where your current dir is, there must also be the root of the CSAR you're trying to initialize/deploy.  So if you are in `tmp/http` and you are running `opera init http.zip` opera searches for the linked playbooks in the current working dir but not in the actual root of the CSAR (`.opera/csars/csar`) which would be right.

So in order to overcome this issue we will have to refactor some code so that opera will use `.opera/csars/csar` as the working directory when initializing/deploying/undeploying compressed CSAR files. Do you agree @matejart?

**Comment by anzoman** [comment]
This has been fixed with #82. Now when we try to deploy the [http.zip](https://github.com/xlab-si/xopera-opera/files/4973145/http.zip) CSAR, opera proceeds without any errors:

```console
(.venv) user@ubuntu:~/orchestration$ opera init http.zip 
CSAR was initialized
(.venv) user@ubuntu:~/orchestration$ opera deploy
[Worker_0]   Deploying Workstation_0
[Worker_0]   Deployment of Workstation_0 complete
[Worker_0]   Deploying DockerEngine_0
[Worker_0]   Deployment of DockerEngine_0 complete
[Worker_0]   Deploying DeploymentTestAgent_0
[Worker_0]     Executing create on DeploymentTestAgent_0
```

The current pre-release where this is available for testing is here: https://test.pypi.org/project/opera/0.5.10.dev11/. The feature will be fully available within the upcoming opera production release (probably **0.5.10**), therefore this issue can be closed.
